GVRf/Framework/jni/shaders/posteffect/custom_post_effect_shader.cpp

```diff
@@ -137,6 +139,16 @@ void CustomPostEffectShader::render(
         glUniform1i(u_texture_, texture_index++);
     }
 
+    if (u_projection_matrix_ != -1) {
+        glm::mat4 view = camera->getViewMatrix();
+        glm::mat4 proj = glm::mat4(
```

Comment:
Hi,

On Tue, May 12, 2015 at 05:34:06PM -0700, thomasflynn wrote:
> > @@ -137,6 +139,16 @@ void CustomPostEffectShader::render(
> >          glUniform1i(u_texture_, texture_index++);
> >      }
> >  
> > +    if (u_projection_matrix_ != -1) {
> > +        glm::mat4 view = camera->getViewMatrix();
> > +        glm::mat4 proj = glm::mat4(
> 
> the rest of the code looks pretty straight-forward, this line is the only one I have a question on so far.  Why this extra step of creating another mat4?
> 

The view matrix needs to be transposed in order to represent the
projection matrix. As the latter is the one I provide to the shaders,
the operation is needed.